src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -7586,6 +7786,11 @@ int RGWRados::set_olh(const DoutPrefixProvider *dpp, RGWObjectCtx& obj_ctx,
     return -EIO;
   }
 
+  // exit early if we're skipping the olh update and just updating the index
+  if (skip_olh_obj_update) {
+    return 0;
```

Comment:
ok, this is a clever way to reuse `set_olh()` without modifying the olh head object 👍

Reply:
note that `set_olh()` still calls `olh_init_modification()` which writes a `RGWOLHPendingInfo` xattr to the olh head object

Reply:
> note that `set_olh()` still calls `olh_init_modification()` which writes a `RGWOLHPendingInfo` xattr to the olh head object

I'm guessing you don't think that's an issue, but let me know if you think it is.

Reply:
my slight preference was to avoid modifying the olh head objects during this index recovery. this adds 'pending' xattrs that won't be cleaned up until a Get/HeadObject request, if ever

the alternative would be to call `bucket_index_link_olh()` directly instead of going through `set_olh()`. i think this was the initial approach that we discussed in slack

feel free to disregard if you think that makes it too complicated

Reply:
Thanks, @cbodley. Now using `bucket_index_link_olh()` and `bucket_index_trim_olh_log()` and it's passing my tests.